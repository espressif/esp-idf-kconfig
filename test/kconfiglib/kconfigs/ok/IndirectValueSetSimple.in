mainmenu "Test simple indirect set"

    comment "Test indirect set for non bools (set option)"

    config A
        bool
        default y
        set B=42

    config B # Should be 42, A is y
        int "b"
        default 0

    config C # Won't be in the output sdkconfig as it has no prompt and is n
        bool
        default n
        set D=43

    config D # Should be 0, C is n
        int "d"
        default 0

    config E # Won't be in the output sdkconfig as it has no prompt and is n
        bool
        set F=44

    config F # Should be 0, E is implicitly n
        int "f"
        default 0

    config G
        bool
        set H=45

    config G_SELECTOR
        bool
        default y
        select G

    config H # Should be 45, G is selected, thus y
        int "h"
        default 0


    comment "Test indirect set for non bools (set default)"

    config I
        bool
        default y
        set default J=42

    config J # Should be 42, I is y
        int "j"
        default 0

    config K # Won't be in the output sdkconfig as it has no prompt and is n
        bool
        default n
        set default L=43

    config L # Should be 0, K is n
        int "l"
        default 0

    config M # Won't be in the output sdkconfig as it has no prompt and is n
        bool
        set default F=44

    config N # Should be 0, M is implicitly n
        int "n"
        default 0

    config O
        bool
        set default P=45

    config O_SELECTOR
        bool
        default y
        select O

    config P # Should be 45, O is selected, thus y
        int "p"
        default 0
