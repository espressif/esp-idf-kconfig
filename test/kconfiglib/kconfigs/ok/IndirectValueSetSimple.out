#
# Automatically generated file. DO NOT EDIT.
# Espressif IoT Development Framework (ESP-IDF)  Project Configuration
#

#
# Test indirect set for non bools (set option)
#
# default:
CONFIG_A=y
# default:
CONFIG_B=42
# default:
CONFIG_D=0
# default:
CONFIG_F=0
# default:
CONFIG_G=y
# default:
CONFIG_G_SELECTOR=y
# default:
CONFIG_H=45

#
# Test indirect set for non bools (set default)
#
# default:
CONFIG_I=y
# default:
CONFIG_J=42
# default:
CONFIG_L=0
# default:
CONFIG_N=0
# default:
CONFIG_O=y
# default:
CONFIG_O_SELECTOR=y
# default:
CONFIG_P=45
